connexion/utils.py

```diff
@@ -131,7 +131,9 @@ def is_json_mimetype(mimetype):
     :rtype: bool
     """
     maintype, subtype = mimetype.split('/')  # type: str, str
-    return maintype == 'application' and (subtype == 'json' or subtype.endswith('+json'))
+    return maintype == 'application' and (subtype == 'json'
+                                          or subtype == 'json; charset=utf-8'
+                                          or subtype.endswith('+json'))
```

Comment:
Would using `subtype.containes('json')` be enough to better cover this?

Reply:
If doing this we would accept invalid content-type? like `Content-type: application/json; charset=utf-83w9fmsefew0`, 

Also from https://stackoverflow.com/questions/9254891/what-does-content-type-application-json-charset-utf-8-really-mean and http://www.ietf.org/rfc/rfc4627.txt. 
We can see `Content-type: application/json; charset=utf-8 ` <=> `Content-type: application/json`, otherwise we would have ti support more encoding.

Reply:
For the RFC4627, you didn't even highlight a specific passage, but I still checked and saw 2 things:

     JSON text SHALL be encoded in Unicode.  The default encoding is
     UTF-8.

and 


      JSON may be represented using UTF-8, UTF-16, or UTF-32.  When JSON
      is written in UTF-8, JSON is 8bit compatible.  When JSON is
      written in UTF-16 or UTF-32, the binary content-transfer-encoding
      must be used.

So it is indeed possible to not only send UTF-8.

But in our case, I agree it's fine that utf-8 is enough, as it's the default for now.

How would the code deal with `application/json;charset=utf-8`?

Reply:
Based on our discussion, what do you think of this implementation?
```
# Reference is https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types
# Where we can see MIME type can have optional parameter
# We can have a space or not in charset
def is_json_mimetype(mimetype):
    """
    :type mimetype: str
    :rtype: bool
    """
    simple_mime_type, optional_parameter = mimetype.split(";") if \
        ";" in mimetype else \
        (mimetype, None)
    maintype, subtype = simple_mime_type.split('/')  # type: str, str

    return maintype == 'application' and \
           (
                   subtype == 'json' or
                   subtype.endswith('+json')
           ) and \
           (
                   optional_parameter is None or
                   optional_parameter.strip() == "charset=utf-8"
           )


```

where

```
import unittest

from mimetype import is_json_mimetype


class TestUtils(unittest.TestCase):

    def test_is_json_mimetype_nominal(self):

        print(is_json_mimetype("application/json"))
        self.assertTrue(is_json_mimetype("application/json"))
        self.assertTrue(is_json_mimetype("application/json; charset=utf-8"))
        self.assertTrue(is_json_mimetype("application/json;charset=utf-8"))
        self.assertTrue(is_json_mimetype("application/anything+json;charset=utf-8"))
        self.assertTrue(is_json_mimetype("application/anything+json"))

        self.assertFalse(is_json_mimetype("application/xml"))
        self.assertFalse(is_json_mimetype("application/json;charset=utf-dfdfd"))

````



`

Reply:
I think this is better. (Just remove the print in the test, and maybe simplify the return value of the function, as it's a 9-lines return value).

Go ahead!

Reply:
Thanks for your suggestion I have updated accordingly.
I made some test with a sample server to check behavior here:
https://github.com/scoulomb/connexion/commit/fc315835c1de37707835f9d1f8043588e255d35e